daffodil-lib/src/main/scala/org/apache/daffodil/xml/XMLUtils.scala

```diff
@@ -702,6 +706,17 @@ object XMLUtils {
     res
   }
 
+  def isXMLIllegalChar(i: Int) = {
+    // #x9 | #xA | #xD | [#x20-#xD7FF] | [#xE000-#xFFFD] | [#x10000-#x10FFFF]
+    i match {
+      case 0x9 | 0xA | 0xD => false
+      case z if (z >= 0x20 && z <= 0xD7FF) => false
+      case z if (z >= 0xE000 && z <= 0xFFFD ) => false
+      case z if (z >= 0x10000 && z <= 0x10FFFF) => false
+      case _ => true
+    }
+  }
+
```

Comment:
Doesn't look like this is used anywhere? Is this jus a helpful tool that we may need in the future, or was the intentino to may replace some of the above code with it?

Might also be worth modifying the comment to say "legal characters: ..." or something. Since the function is called isXMLIllegalChars I assumed that comment was the list of illegal charactes and was a bit confused.

Also, just an FYI, I was curious to see how this match/catch compiled to bytecode compared to if-statements, and they end up exactly the same. Not too surprised, but good to know.

Reply:
Yes, I'm curious too.

Reply:
I will remove this.

This is me once again not remembering that I already wrote something long ago. I started writing it again, then found what I needed. 

I may add some scaladoc to these escapers, pua mappers, etc. to explain their usage, what they do (and don't do).